apps/hash-api/src/graph/knowledge/system-types/file.ts

```diff
@@ -63,12 +84,39 @@ const generateCommonParameters = async (
       ),
     };
   } else if (fileEntityCreationInput) {
+    const { entityTypeId: specifiedEntityTypeId } = fileEntityCreationInput;
+
+    const entityTypeIdByMimeType = getEntityTypeIdForMimeType(mimeType);
+
+    let entityTypeId: VersionedUrl;
+
+    if (specifiedEntityTypeId) {
+      /**
+       * If there is a mime entity type ID and it is not the same
+       * as the specified entity type ID, override it. Otherwise,
+       * use the specified entity type ID.
```

Comment:
I thought we would allow users to specify their own `entityTypeId` if they wanted, and we would only override it if it was a `hash` system type

Reply:
We can do this, but this means we have to add the logic in the FE for detecting the file type (so that the correct `entityTypeId` is passed here) which I thought we didn't want. If we do, I can move it there.

Reply:
Detecting it where / why? If they're coming from a HASH system type then we detect the most appropriate specific type here. If they're passing a non-HASH `entityTypeId` then let them use it, no detection is necessary

Reply:
The entity type upload tab currently specifies an `entityTypeId`. If we want to let it continue specifying the `entityTypeId`, then we either need to override it here based on the mime type, or override it in the frontend based on the mime type. Otherwise when you upload a DOCX file in the generic "file" uploader in the frontend, you're not going to end up with a file entity of type `DOCX`, but a file entity of type `File`.

Reply:
Sorry I misunderstood what you were saying. I've made it so that only if a HASH entity type is specified it can be overriden by the mime type, in https://github.com/hashintel/hash/pull/3929/commits/613542848ba315a5ca99c29ede34ae55bacc4190